### PR TITLE
Fix migrate command

### PR DESCRIPTION
**Pull request by Loonride**
**What kind of change does this PR introduce?**

fix/tests

**Did you add tests for your changes?**

Yes (still need to add tests for runPrettier)

I'm not exactly sure how the migrate command is supposed to work. Could someone give an example of a webpack config that it should make modifications to?

**If relevant, did you update the documentation?**

No

**Summary**

@jamesgeorge007 Sorry for taking over, just couldn't debug `runPromptWithAnswers` without using your changes. It seems that execa does not like an empty `input` string option.

- Fixes `runPromptWithAnswers`
- Fixes migrate
- Fixes issues with `runPrettier`

**Does this PR introduce a breaking change?**

Slightly with how `runPrettier` works

**Other information**

### CODE REVIEW

**Comment by anshumanv** [comment]
@Loonride one test unusually still fails, doesn't show any log, can you debug?

**Comment by jamesgeorge007** [comment]
> I'm not exactly sure how the migrate command is supposed to work. Could someone give an example of a webpack config that it should make modifications to?

You can find everything related to `migrate` command [here](https://github.com/webpack/webpack-cli/blob/next/MIGRATE.md)

File: test/utils/test-utils.js [hunk 1]
```diff
@@ -67,17 +67,24 @@ function runWatch({ testCase, args = [], setOutput = true, outputKillStr = 'Time
...
 
-    const webpackProc = execa(WEBPACK_PATH, argsWithOutput, {
+    const options = {
         cwd,
         reject: false,
         stdio: ENABLE_LOG_COMPILATION ? 'inherit' : 'pipe',
-    });
+    };
+
+    // some tests don't work if the input option is an empty string
```
Comment (jamesgeorge007) [inline test/utils/test-utils.js:82]: Good catch 👏

**Comment by Loonride** [comment]
It seems sending a SIGKILL to the process at the end of `runPromptWithAnswers` made it more stable :woman_shrugging:

**Comment by evilebottnawi** [comment]
hm, ci is not stable in some cases, it is strange

**Comment by Loonride** [comment]
> hm, ci is not stable in some cases, it is strange

whats confusing is it always either seems like a large group of them fail, or none of them fail :thinking:

**Comment by rishabh3112** [comment]
> whats confusing is it always either seems like a large group of them fail, or none of them fail 🤔
Others are cancelled when one of them fails. the macOS Node v13 container is the failing one everywhere :(

**Comment by Loonride** [comment]
> Others are cancelled when one of them fails. the macOS Node v13 container is the failing one everywhere :(

Oh that makes more sense thanks

**Comment by evilebottnawi** [comment]
I think we have racing

**Comment by Loonride** [comment]
Let me try a few things with how `runPromptWithAnswers` works.

**Comment by Loonride** [comment]
@evilebottnawi 4 test runs good in a row, I think stabilization worked!

What I did was forced `runPromptWithAnswers to wait for new stdout before writing the next answer to stdin

**Comment by evilebottnawi** [comment]
Good job!

**Pull request merged**